src/lib.rs

```diff
@@ -50,6 +50,7 @@
 #![doc(test(attr(allow(unused_variables), deny(warnings))))]
 #![cfg_attr(not(feature = "std"), no_std)]
 #![cfg_attr(all(feature = "simd_support", feature = "nightly"), feature(stdsimd))]
+#![cfg_attr(all(feature = "partition_at_index", feature = "nightly"), feature(slice_partition_at_index))]
```

Comment:
This however makes separate use of `partition_at_index` pointless — I think unlike `simd_support` we should not depend on `nightly` here. As a bonus, this makes it impossible to use `partition_at_index` on stable compilers.

Reply:
I still don't understand what the advantage of a separate `partition_at_index` is.